public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -85,29 +77,24 @@ export class LokiDatasource {
     const queries = queryTargets.map(target => this._request('/api/prom/query', target));
 
     return Promise.all(queries).then((results: any[]) => {
-      const allStreams: LogsStream[] = [];
+      const series: SeriesData[] = [];
 
       for (let i = 0; i < results.length; i++) {
         const result = results[i];
-        const query = queryTargets[i];
-
-        // add search term to stream & add to array
         if (result.data) {
           for (const stream of result.data.streams || []) {
-            stream.search = query.regexp;
-            allStreams.push(stream);
+            const seriesData = logStreamToSeriesData(stream);
+            seriesData.name = 'logs';
+            seriesData.meta = {
+              search: queryTargets[i].regexp,
+              limit: this.maxLines,
+            };
```

Comment:
For now added these properties into meta field.

Reply:
think that should be a good use of the meta field

Reply:
+1 -- these also seem reasonable to put in the typescript defs rather than just `any`

Reply:
Fixed